api/handlers/node.go

```diff
@@ -65,6 +96,84 @@ func (h *Node) Identity(w http.ResponseWriter, r *http.Request) error {
 
 func (h *Node) Peers(w http.ResponseWriter, r *http.Request) error {
 	peers := h.Network.Peers()
+	resp := h.peers(peers)
+	return api.Render(w, r, resp)
+}
+
+func (h *Node) Topics(w http.ResponseWriter, r *http.Request) error {
+	allpeers, peerbytpc := h.TopicIndex.PeersByTopic()
+	alland := AllPeersAndTopicsJSON{}
+	tpcs := []topicIndexJSON{}
+	for topic, peers := range peerbytpc {
+		tpcs = append(tpcs, topicIndexJSON{TopicName: topic, Peers: peers})
+	}
+	alland.AllPeers = allpeers
+	alland.PeersByTopic = tpcs
+
+	return api.Render(w, r, alland)
+}
+
+func (h *Node) Health(w http.ResponseWriter, r *http.Request) error {
+	ctx := context.Background()
+	resp := healthCheckJSON{
+		BeaconConnectionHealthStatus:    Healthy.String(),
+		ExecutionConnectionHealthStatus: Healthy.String(),
+		EventSyncHealthStatus:           Healthy.String(),
+	}
+	// check ports being used
+	addrs := h.Network.ListenAddresses()
+	for _, addr := range addrs {
+		if addr.String() == "/p2p-circuit" || addr.Decapsulate(multiaddr.StringCast("/ip4/0.0.0.0")) == nil {
+			continue
+		}
+		resp.LocalPortsListening = addr.String()
+	}
+	// check consensus node health
+	err := h.NodeProber.CheckBeaconNodeHealth(ctx)
+	if err != nil {
+		resp.BeaconConnectionHealthStatus = fmt.Sprintf("%s: %s", NotHealthy, err.Error())
+	}
+	// check execution node health
+	err = h.NodeProber.CheckExecutionNodeHealth(ctx)
+	if err != nil {
+		resp.ExecutionConnectionHealthStatus = fmt.Sprintf("%s: %s", NotHealthy, err.Error())
+	}
+	// check event sync health
+	err = h.NodeProber.CheckEventSyncerHealth(ctx)
+	if err != nil {
+		resp.EventSyncHealthStatus = fmt.Sprintf("%s: %s", NotHealthy, err.Error())
+	}
+	// check peers connection
+	var activePeerCount int
+	peers := h.Network.Peers()
+	for _, p := range h.peers(peers) {
+		if p.Connectedness == "Connected" {
+			activePeerCount++
+		}
+	}
+	switch {
+	case activePeerCount >= healthyPeersAmount:
+		resp.PeersHealthStatus = fmt.Sprintf("%s: %d  peers are connected", Healthy, activePeerCount)
+	case activePeerCount > 0 && activePeerCount < healthyPeersAmount:
+		resp.PeersHealthStatus = fmt.Sprintf("%s: less than %d peers are connected", NotHealthy, healthyPeersAmount)
+	case activePeerCount == 0:
+		resp.PeersHealthStatus = fmt.Sprintf("%s: %s", NotHealthy, "error: no peers are connected")
+	}
```

Comment:
I would break this into two. one that says GOOD/BAD (so status only), and one that says the number, or maybe reason.